app/assets/javascripts/components/story/StoryNotes.js

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import Note from 'components/notes/Note';
+import NoteForm from 'components/notes/NoteForm'
+
+class StoryNotes extends React.Component {
+  constructor() {
+    super();
+
+    this.renderNotes = this.renderNotes.bind(this);
+    this.renderStoryNotes = this.renderStoryNotes.bind(this);
+  }
+
+  render() {
+    return this.props.isNew ? null : this.renderStoryNotes();
+  }
+
+  renderStoryNotes() {
+    const { notes, disabled } = this.props;
+    return (
+      <div className="form-group">
+        <label htmlFor="notes"> { I18n.t('story.notes') } </label>
+        <div className="notelist">
+          { this.renderNotes() }
+        </div>
+        { !disabled && <NoteForm note={notes.last()} /> }
+      </div>
+    );
+  }
+
+  renderNotes() {
+    const { notes, disabled, handleDelete } = this.props;
+    const noteList = (disabled) ? notes : notes.slice(0, -1);
+    return noteList.map((note, i) =>
+      <Note
+        note={note}
+        disabled={disabled}
+        handleDelete={handleDelete}
+        key={i}
```

Comment:
Never use the array index as key for React components. The `key` props is used by React to track what component corresponds to wich item on an array, if you pass the index as the key and then remove the first item of the array (for example) the element with the key `0` will be different after the component re-renders but with the same key value of the previous one. For cases like that prefer to use the id of the record. I think you can do: `key={note.get('id')}`. You can read more about it here: https://medium.com/@robinpokorny/index-as-a-key-is-an-anti-pattern-e0349aece318